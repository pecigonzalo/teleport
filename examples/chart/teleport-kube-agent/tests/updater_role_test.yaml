suite: Updater Role
templates:
  - updater/role.yaml
tests:
  #
  # Basic tests
  #
  - it: does not create a Role when updater.enabled is false (default)
    asserts:
      - hasDocuments:
          count: 0
  - it: creates a Role when updater.enabled is true
    values:
      - ../.lint/updater.yaml
    asserts:
      - containsDocument:
          kind: Role
          apiVersion: rbac.authorization.k8s.io/v1
          name: RELEASE-NAME-updater
          namespace: NAMESPACE
  - it: does not create a Role when updater.enabled is true but rbac creation is disabled
    values:
      - ../.lint/updater.yaml
    set:
      rbac:
        create: false
    asserts:
      - hasDocuments:
          count: 0

  #
  # Catch-all content test
  #
  - it: sets the correct role rules
    values:
      - ../.lint/updater.yaml
    asserts:
      - matchSnapshot:
          path: rules
