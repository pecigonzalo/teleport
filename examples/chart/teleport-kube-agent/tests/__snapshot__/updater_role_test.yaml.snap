sets the correct role rules:
  1: |
    - apiGroups:
      - ""
      resources:
      - pods
      verbs:
      - get
      - watch
      - list
      - delete
    - apiGroups:
      - ""
      resources:
      - pods/status
      verbs:
      - get
      - watch
      - list
    - apiGroups:
      - ""
      resources:
      - secrets
      verbs:
      - watch
      - list
    - apiGroups:
      - ""
      resourceNames:
      - RELEASE-NAME-shared-state
      resources:
      - secrets
      verbs:
      - get
    - apiGroups:
      - ""
      resources:
      - events
      verbs:
      - create
      - patch
    - apiGroups:
      - apps
      resources:
      - deployments
      - statefulsets
      - deployments/status
      - statefulsets/status
      verbs:
      - get
      - watch
      - list
    - apiGroups:
      - apps
      resourceNames:
      - RELEASE-NAME
      resources:
      - deployments
      - statefulsets
      verbs:
      - update
    - apiGroups:
      - coordination.k8s.io
      resources:
      - leases
      verbs:
      - create
    - apiGroups:
      - coordination.k8s.io
      resourceNames:
      - RELEASE-NAME
      resources:
      - leases
      verbs:
      - get
      - update
